### DISCUSSION TITLE
23 July - Dev Session (Strategy Session)

### DISCUSSION THREAD

**Discussion by tpike3** [mesa/mesa#1367] [General]
When: 23 July 2000 UTC+8/1300 CET/1200 GMT/0800 ET

Where: [via Matrix](https://matrix.to/#/#mesa-dev:matrix.org)

Proposed Agenda: 
- Discuss Mesa dynamics such as: 
      - What does the mesa ecosystem look like?
      - Long term goals for sustainment (discuss NUMFOCUS Small Business Grant)
      - Areas to nurture relationships
 
@jackiekazil @rht @Corvince @wang-boyu @aiwhoo @nunezco2 @EwoutH  Please feel free to add or takeaway from this agenda

**Comment by tpike3** [comment]
**Notes**

Mesa Strategy Session started with a discussion on the use of [Mesa for a COVID-19 model](https://github.com/ncsa/COVID19-mesa) led by @nunezco2. This model has introduced a lot of people to Mesa and also has been used on a supercomputer. Recently it was part of a [Computational Social Science Society of Americas](https://computationalsocialscience.org/) talk. We are anxiously awaiting the posting of the video. 
 
2 themes came up during the discussion: 

**Develop the Mesa Community**:
How do we grow a more robust and resilient Mesa community? This becomes particularly important as we have engaged with [NUMFOCUS](https://numfocus.org/) and can request small grants, but need a community of developers we can activate to use these funds if we receive a grant. The area of most success has been with academic institutions so we want to continue developing along this vein. This question also brought up the discussion of how do open source libraries grow and develop and so we want to research along these lines. 

Some suggested avenues: 
- continue to engage with institutions we have relationships with TUDelft, U. Mary Washington, George Mason and developing with U. of Illinois Urbana-Champagne. 
- engage more heavily with [NUMFOCUS](https://numfocus.org/) and learn from other successful projects.
- engage with organizations such as [RSE](https://society-rse.org/#section-4)     

What we need for this is metrics and timelines so we know how we are doing. 

**Mesa Strategic Development**
We talked about the challenge of building an ABM ecosystem so users can rapidly build models to aid decision making and understanding. An exemplar of this in industry is [#Hash](https://hash.ai/). Through this it came up that visualizations and facilitating individuals interacting with simulations is critical. 

This led to three areas of recommended development focus, two we have already started working in
1. [Mesa-Geo](https://github.com/projectmesa/mesa-geo) being led by @wang-boyu 
2. Improved Mesa visuals. This has already started and is continuing along two paths, #1370 and #1354  
3. The idea of developing cached results extension, so after a model is developed it can be used to interact with users to they can get rapid results without necessarily the computational overhead of complex models.

What we need for this is timelines and associated goals such as getting a grant to build a specific capability.

**Comment by wang-boyu** [comment]
Regarding community development, I recently watched a video about [open-source strategy at VictoriaMetrics](https://www.youtube.com/watch?v=-DbbIZzFHIY). They mentioned using monitoring tools (e.g., [GrimoireLab](https://chaoss.community/software/#user-content-grimoirelab), [Cauldron](https://cauldron.io)) to build dashboards on project metrics. Is this worth considering for us as well?

**Reply by rht** [reply]
Which one among the metrics tools is simpler to setup?

**Reply by wang-boyu** [reply]
Looks like `GrimoireLab` is set of tools that we'll need to deployment by ourselves. There are free (and paid) services like [Cauldron](https://cauldron.io/) and [LFX Insights](https://insights.lfx.linuxfoundation.org) built on top of it. Perhaps we can try one of the free services?

**Reply by rht** [reply]
https://cauldron.io/project/6853?from_date=2021-09-30&to_date=2022-09-30&tab=overview
